framework/src/modules/interoperability/sidechain/module.ts

```diff
@@ -195,4 +211,184 @@ export class SidechainInteroperabilityModule extends BaseInteroperabilityModule
 	public async init(_args: ModuleInitArgs) {
 		this._mainchainRegistrationCommand.addDependencies(this._validatorsMethod);
 	}
+
+	// @see https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#mainchain
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async initGenesisState(ctx: GenesisBlockExecuteContext): Promise<void> {
+		const genesisBlockAssetBytes = ctx.assets.getAsset(MODULE_NAME_INTEROPERABILITY);
+		if (!genesisBlockAssetBytes) {
+			return;
+		}
+
+		const genesisInteroperability = codec.decode<GenesisInteroperability>(
+			genesisInteroperabilitySchema,
+			genesisBlockAssetBytes,
+		);
+
+		validator.validate<GenesisInteroperability>(
+			genesisInteroperabilitySchema,
+			genesisInteroperability,
+		);
+
+		await this._verifyChainInfos(ctx, genesisInteroperability);
+	}
```

Comment:
Also, if I am not missing something, we should put here the "Genesis State Processing" part. 

I don't see the genesis state finalization part either.